modules/src/core/ics03_connection/handler/verify.rs

```diff
@@ -42,12 +45,7 @@ pub fn verify_proofs(
 
     // If a consensus proof is attached to the message, then verify it.
     if let Some(proof) = proofs.consensus_proof() {
-        Ok(verify_consensus_proof(
-            ctx,
-            connection_end,
-            proofs.height(),
-            &proof,
-        )?)
+        Ok(verify_consensus_proof(ctx, height, connection_end, &proof)?)
```

Comment:
Unrelated to this PR, but I think we're only verifying the consensus proof if we receive (a non-empty) one in the IBC message, so this can be a security problem. I think the root cause of this issue is the way `Proofs` is modelled, due to which we no longer know if this function was called for a message that required consensus proof verification or not. One way to solve this problem would be to model `Proofs` as an enum - I'll open an issue for this. Can be addressed in a separate PR.

Reply:
https://github.com/informalsystems/ibc-rs/issues/1696